build_debian.sh

```diff
@@ -629,6 +630,56 @@ then
 
 fi
 
+# #################
+#   secure boot 
+# #################
+if [[ $SECURE_UPGRADE_MODE == 'dev' || $SECURE_UPGRADE_MODE == "prod" && $SONIC_ENABLE_SECUREBOOT_SIGNATURE != 'y' ]]; then
+    # note: SONIC_ENABLE_SECUREBOOT_SIGNATURE is a feature that signing just kernel, 
+    # SECURE_UPGRADE_MODE is signing all the boot component including kernel.
+    # its required to do not enable both features together to avoid conflicts.
+    echo "Secure Boot support build stage: Starting .."
+
+    # debian secure boot dependecies
+    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install      \
+        shim-unsigned \
+        grub-efi
+    
+    if [ ! -f $SECURE_UPGRADE_DEV_SIGNING_CERT ]; then
+        echo "Error: SONiC SECURE_UPGRADE_DEV_SIGNING_CERT=$SECURE_UPGRADE_DEV_SIGNING_CERT key missing"
+        exit 1
+    fi
+
+    if [[ $SECURE_UPGRADE_MODE == 'dev' ]]; then
+        # development signing & verification 
+
+        if [ ! -f $SECURE_UPGRADE_DEV_SIGNING_KEY ]; then
+            echo "Error: SONiC SECURE_UPGRADE_DEV_SIGNING_KEY=$SECURE_UPGRADE_DEV_SIGNING_KEY key missing"
+            exit 1
+        fi
+
+        sudo ./scripts/signing_secure_boot_dev.sh -a $CONFIGURED_ARCH \
+                                                  -r $FILESYSTEM_ROOT \
+                                                  -l $LINUX_KERNEL_VERSION \
+                                                  -c $SECURE_UPGRADE_DEV_SIGNING_CERT \
+                                                  -p $SECURE_UPGRADE_DEV_SIGNING_KEY
+    elif [[ $SECURE_UPGRADE_MODE == "prod" ]]; then
```

Comment:
>elif [](http://example.com/codeflow?start=4&length=4)

We should minimize the difference between two modes.
1. share OUTPUT_SEC_BOOT_DIR
2. verify EFI files and kernel modules
3. verify vmlinuz file

Reply:
well, the verification is done by the same script name `./scripts/secure_boot_signature_verification.sh`, but it's not exactly the same flow.

in development, the script `scripts/signing_secure_boot_dev.sh` is doing the verification after each local signing, so if there is some failure the build will stop immediately and not continue the signing of another component.

in prod, the sign happened in a remote server so all the signing logic coded in the script `SECURE_UPGRADE_PROD_SIGNING_TOOL` will finish, and then the verification step will start doing the verification of all the components.

Reply:
It seems that the verification method could be unified. So the if-elif block will be smaller. If you really want "verification after each local signing, so if there is some failure the build will stop immediately", you may have duplicated verification, and it does not harm much.

Reply:
I not sure 100% its possible, probably the best it's to schedule a meeting to understand exactly the details.
Or maybe if it possible, can you approve this PR as is, and I will schedule a meeting and create an additional PR to review your suggestion?
I preferred to do not postpone this merge only in result of a verification step, if its ok from your side as well.